src/interface.js

```diff
@@ -15,6 +15,16 @@ export default function(Target) {
   // Create a new instance of the `Runner`, passing in the current object.
   Target.prototype.then = function(/* onFulfilled, onRejected */) {
     const result = this.client.runner(this).run()
+
+    if (this.client.config.asyncStackTraces) {
+      result.catch((err) => {
+        err.originalStack = err.stack
```

Comment:
Might be wrong, but shouldn't it also throw here?

Reply:
I did try both and it seems we don't want to throw.
Without throw, when I run this code:
```javascript
const test = async () => {
  try {
    await db('jijoi').select()
  } catch (err) {
    console.log('caught')
  }
  console.log('can be caught')
}

test()
```

I get this:
```
caught
can be caught
```

but when I add the throw there:
```
caught
can be caught
Unhandled rejection error: relation "jijoi" does not exist
    at test (/home/capaj/git_projects/tests/knex.js:18:11)
    at Object.<anonymous> (/home/capaj/git_projects/tests/knex.js:25:1)
    at Module._compile (module.js:649:30)
    at Object.Module._extensions..js (module.js:660:10)
    at Module.load (module.js:561:32)
    at tryModuleLoad (module.js:501:12)
    at Function.Module._load (module.js:493:3)
    at Function.Module.runMain (module.js:690:10)
    at startup (bootstrap_node.js:194:16)
    at bootstrap_node.js:666:3
```

There's that unhandled rejection.

When I don't throw, it behaves well even when I don't try-catch:
```javascript
const test = async () => {
  await db('jijoi').select()
  console.log('does not run')
}

test()
```
outputs:
```
Unhandled rejection error: relation "jijoi" does not exist
    at test (/home/capaj/git_projects/tests/knex.js:17:9)
    at Object.<anonymous> (/home/capaj/git_projects/tests/knex.js:21:1)
    at Module._compile (module.js:649:30)
    at Object.Module._extensions..js (module.js:660:10)
    at Module.load (module.js:561:32)
    at tryModuleLoad (module.js:501:12)
    at Function.Module._load (module.js:493:3)
    at Function.Module.runMain (module.js:690:10)
    at startup (bootstrap_node.js:194:16)
    at bootstrap_node.js:666:3

(node:31738) UnhandledPromiseRejectionWarning: error: relation "jijoi" does not exist
    at test (/home/capaj/git_projects/tests/knex.js:17:9)
    at Object.<anonymous> (/home/capaj/git_projects/tests/knex.js:21:1)
    at Module._compile (module.js:649:30)
    at Object.Module._extensions..js (module.js:660:10)
    at Module.load (module.js:561:32)
    at tryModuleLoad (module.js:501:12)
    at Function.Module._load (module.js:493:3)
    at Function.Module.runMain (module.js:690:10)
    at startup (bootstrap_node.js:194:16)
    at bootstrap_node.js:666:3
(node:31738) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:31738) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Reply:
I think this has got to do with the fact that knex does not return a real promise-it just has methods `then` and `catch` so it's masquerading like a promise, but it's really not.

Reply:
I think @wubzz was right that it should throw. Now this extra catch is just added as an additional branch how result is resolved.

So instead of adding this new `.catch(..)` handler as a separate branch which is then left dangling without having reference to the returned promise, it should be added like this:

```
result = result.catch(...)
```